services/webnn/ort/graph_builder_ort.cc

```diff
@@ -2220,79 +2261,183 @@ GraphBuilderOrt::AddLayerNormalizationOperation(
+
+  std::ranges::sort(axes);
+  std::vector<uint32_t> scale_shape;
+  scale_shape.reserve(axes_size);
+  std::ranges::transform(
+      axes, std::back_inserter(scale_shape),
+      [&input_shape](uint32_t axis) { return input_shape[axis]; });
+
+  std::string scale;
+  std::string bias;
   // Here we only check beginning and ending of the ascending sorted axes,
   // because the blink validation code ensures axes not having duplicated
   // values.
-  // TODO: support inconsecutive axes by emulation -
-  // https://github.com/shiyi9801/chromium/issues/69.
-  if (axes[axes_size - 1] != input_shape.size() - 1 ||
-      axes[0] != input_shape.size() - axes_size) {
-    return NewNotSupportedError(
-        "ONNX LayerNormalization only supports last consecutive dimensions "
-        "as axes.");
-  }
-  uint32_t axis = axes[0];
-  std::string scale;
-  base::CheckedNumeric<uint32_t> checked_scale_size =
-      std::accumulate(input_shape.begin() + axis, input_shape.end(),
-                      base::CheckedNumeric<uint32_t>(1), std::multiplies());
-  if (!checked_scale_size.IsValid()) {
-    return NewNotSupportedError("The size of scale is too large.");
-  }
+  if (axes[axes_size - 1] == input_shape.size() - 1 &&
+      axes[0] == input_shape.size() - axes_size) {
+    ASSIGN_OR_RETURN(scale, CreateOrTransposeScaleForLayerNomalization(
+                                layer_normalization.scale_operand_id,
+                                input_data_type, scale_shape, permutation));
+    inputs.push_back(scale.c_str());
 
-  std::vector<uint32_t> scale_dims;
-  scale_dims.reserve(axes_size);
-  std::ranges::transform(
-      axes, std::back_inserter(scale_dims),
-      [&input_shape](uint32_t axis) { return input_shape[axis]; });
+    if (layer_normalization.bias_operand_id) {
+      bias = GetOperandNameById(layer_normalization.bias_operand_id.value());
+      bias = PrependTranspose(bias, permutation);
+      inputs.push_back(bias.c_str());
+    }
 
-  if (layer_normalization.scale_operand_id) {
-    scale = GetOperandNameById(layer_normalization.scale_operand_id.value());
-    inputs.push_back(scale.c_str());
+    std::vector<ScopedOrtOpAttr> attributes;
+    attributes.reserve(2);
+    attributes.push_back(model_editor_.CreateAttribute(
+        /*name=*/"axis", base::checked_cast<int64_t>(axes[0])));
+    attributes.push_back(model_editor_.CreateAttribute(
+        /*name=*/"epsilon", layer_normalization.epsilon));
+
+    model_editor_.AddNode(kOpTypeLayerNormalization, node, inputs, outputs,
+                          std::move(attributes));
   } else {
+    // Emulation layerNormalization by scale * ((input - mean) / sqrt(variance +
+    // epsilon)) + bias. Calculate mean as follows- reduceOptions = {axes,
+    // keepDimensions: true}; mean = builder.reduceMean(input, reduceOptions).
+    std::string inserted_node =
+        GenerateNextOperationName("inserted_reduceMean_1");
```

Comment:
Concate the `layer_normalization.label` with "inserted_reduceMean_1", do the same for other inserted ops.